lasagne/layers/recurrent.py

```diff
@@ -107,7 +107,12 @@ class CustomRecurrentLayer(MergeLayer):
         :class:`lasagne.layers.Layer` instance which connects input to the
         hidden state (:math:`f_i`).  This layer may be connected to a chain of
         layers, which must end in a :class:`lasagne.layers.InputLayer` with the
-        same input shape as `incoming`.
+        same input shape as incoming, except for the first dimension: When
+        ``precompute_input == True`` (the default),
+        ``input_to_hidden.input_shape[0]`` must be
```

Comment:
Hey, that's the same mistake as before. You can't refer to this as `input_to_hidden.input_shape`, because that would be the input to the last layer of `input_to_hidden`, which could be a deep network. Just say "it", which then refers to "except for the first dimension" of "the same input shape" of the ":class:`lasagne.layers.InputLayer`". Also wouldn't it be easier to name `None` only once instead of twice? (https://github.com/Lasagne/Lasagne/pull/455#discussion_r41174756)